### DISCUSSION TITLE
Feature: Comment string interpolation

### DISCUSSION THREAD

**Discussion by kazad** [kazad/instacalc#42] [Announcements]
Now you can have dynamic comments with `{}`:

`x = 3; //{x + 3}`

And the comment text will be "6". This might be useful for certain types of calcs.

Currently, you have to use variables that have been defined before comment row. (A future feature may be allowing out-of-order execution, even for the comment text... but let's just get things working for now!)

**Comment by hobbyman** [comment]
Just trying out this feature....it does not obey the rounding.

My weird scenario is this:
`// {(-2343.20*-1)-1500}`

and the displayed result is this:
`843.1999999999998`

I have the rounding set to 0.01

BTW, this site is AWESOME!!

**Comment by kazad** [comment]
Thanks for the bug report! Just fixed: https://instacalc.com/56964

There was a bug in detecting when to use "high precision" math (turns out things like "-2343.20*-1)-1500" cause floating point issues in JS).

I'm thinking about how best to apply global rounding settings to the interpolated strings, I can see the case where you want fine-grained control over the display (separate from the 'regular' rounding). I'll have to think that over a bit.

Glad you're enjoying the site! =)

**Comment by EC-SG** [comment]
Round() function currently works interpolated in comments.

Kalid - if you could revise Round() to accept number of digits ala Excel, it might be quick solution to this problem.

Round (number, num_digits)

Such a formula would also allow users to access more decimal places where necessary, overriding global settings only at that point of usage.

**Comment by kazad** [comment]
Great find! I added the "digits" param to `round` a while [back](https://github.com/kazad/instacalc/discussions/62), so this should be possible:

https://instacalc.com/58682

Thanks for the suggestion though!

**Comment by EC-SG** [comment]
Interpolated usage of Round with digits param - 

Inconsistent behavior found...  then temporary possible work-around also found.

https://instacalc.com/58683

![image](https://github.com/user-attachments/assets/51354bbc-c25c-4dca-a698-485d38b704b5)

**Comment by kazad** [comment]
Ah! Thanks. So, what's happening is that `3.5522312,2` is being treated as `3.55223122`, i.e. the 2 is being treated as a comma separator for the number, vs. an argument separator for round. 

writing `round(3.5522312, 2)` (space after comma) or `round((3.5522312), 2)` (parens to separate) should work.

I might need to add some logging messages to help explain this, esp. if you aren't using the European or South Asian style numbers. Multiple arguments with multiple number formats gets tricky, I'll have to think about good ways to disambiguate.

**Reply by EC-SG** [reply]
Thank you very much! 
I had forgotten about accommodating number systems. 😄

**Reply by kazad** [reply]
More than welcome! Yeah, there's a lot of subtle interactions :). Always happy to chat about feature ideas / possible bugs.
